documentation/add/09_Serializable_Interfaces.md

```diff
@@ -0,0 +1,132 @@
+
+1. The object has an empty constructor.
+2. The object has a public interface which allows to define(assign) to an empty object any contents the non-empty object may have, defining in this way any valid non-empty object.
+
+If such assumptions are satisfied, we may define *serializable* objects, which need to overload only handful of class-specific methods, but would immediately have number of very useful generic features. The class diagram describing such object is presented on the **Fig 1**:
+
+<center>![Serialiable. Main Interface](../diagrams/serializable_main_interface.png)</center>
+
+**Fig 1:** Main interface and methods of **serializable** class.
+
+The parent class, presented on the **Fig 1**, allows easy construction and maintenance of standardized serializable objects by overloading the following abstract methods:
+
+**Table 1: Main abstact methods requesing overloading (definition)**
+
+| Method | Description |
+|-----|---------|
+|  classVersion | returns the number, which describes current class version. E.g. 1 - for version 1|
+| saveableFields | returns list of property names, defining the class public interface |
+
+As soon as these properties are defined, one can use other class methods to serialize/deserialize objects and convert objects to/from structure. The main features of the available methods are summarized in the Tables 2:
+** Table 2 Main methods, available for serializable objects:**
+
+| Method | Description |
+|-----|---------|
+|  `struct = to_bare_struct(obj)` | returns structure, with the field names equal to the names returned by `saveableFields` method and values, defined by the values of these fields. If the values of fields are the serializable objects themselves, they are converted into correspondent structures recursively |
+| `obj = from_bare_struct(obj,data)` | Reverse of `to_bare_struct` method. Sets the values of the public properties of the serializable interface from the structure, obtained from `to_bare_struct` method |
+| `data = to_struct(obj)` | converts `obj` or array of `obj` into the `data` structure using `to_bare_struct` method and adds complimentary information about the class name and the size and shape of the `obj` array if relevant. |
+| *Static:* | |
+| `obj = from_struct(data)` | Reverse of `to_struct` method, recovering the object or array of objects from the `data`, generated by the `to_struct` method |
+
+Standard `loadobj/saveobj` and `serialize/deserialize` methods use the data obtained from the methods described in **Table 2** or the methods themselves to provide/use input for/from Matlab save/load methods or serialize/deserialize methods to save/load data to/from `.mat` files or transform the objects from/to linear arrays of bytes for communications over serial pipes. The `serialize/deserialize` routines use Herbert-defined `serialize/deserialize` methods over the structures obtained using `to_struct/from_struct` methods.
```

Comment:
Not clear about the last sentence here. I think you want to say that either we use to_struct/from_struct and put/get a struct, or we use serialize/deserialize and get a serialization string. Iam not clear why one would want to use serialise/deserialise

Reply:
you would use `serialize/deserialize` to send object over MPI pipe